dash-pipeline/bmv2/dash_pipeline.p4

```diff
@@ -268,6 +268,34 @@ control dash_ingress(
         }
     }
 
+    action set_src_tag(tag_map_t tag_map) {
+        meta.src_tag_map = tag_map;
+    }
+
+    @name("src_tag|dash_tag")
+    table src_tag {
+        key = {
+            meta.src_ip_addr : lpm @name("meta.src_ip_addr:sip");
+        }
+        actions = {
+            set_src_tag;
+        }
+    }
+
+    action set_dst_tag(tag_map_t tag_map) {
+        meta.dst_tag_map = tag_map;
+    }
+
+    @name("dst_tag|dash_tag")
```

Comment:
I am not sure how the SAI API is derived from the  P4 code, but I noticed that we have separate APIs to create src tag vs dst tag in sai_dash_tag_api_t
why is this the case? this adds a lot of redundancy.

we should be able to define tags and then use those for src or destination.

this would be similar to ACL groups: we just create a group and then use is for ingress or egress.